src/WallpaperEngine/Render/CVideo.cpp

```diff
@@ -0,0 +1,30 @@
+#include "CVideo.h"
+
+using namespace WallpaperEngine;
+
+using namespace WallpaperEngine::Render;
+
+CVideo::CVideo (Core::CVideo* video, irr::video::IVideoDriver* driver) :
+    CWallpaper (video, Type),
+    m_driver (driver)
+{
+    int width = driver->getScreenSize ().Width;
+    int height = driver->getScreenSize ().Height;
+    video->initFrames (width, height);
+    m_frameImage = m_driver->createImage (irr::video::ECOLOR_FORMAT::ECF_R8G8B8,
+                irr::core::dimension2du(width, height));
+}
+
+void CVideo::renderWallpaper ()
```

Comment:
This should be an override of the render function of irr::scene::ISceneNode as I said on another comment